src/VisualStudio/Core/Def/Implementation/Library/ObjectBrowser/AbstractObjectBrowserLibraryManager.cs

```diff
@@ -510,27 +510,28 @@ protected override bool TryExec(Guid commandGroup, uint commandId)
         {
             try
             {
-                // Let the presented know we're starting a search.  It will give us back
-                // the context object that the FAR service will push results into.
+                // Let the presented know we're starting a search.  It will give us back the context object that the FAR
+                // service will push results into.  This operation is not externally cancellable.  Instead, the find
+                // refs window will cancel it if another request is made to use it.
                 var context = presenter.StartSearch(
-                    EditorFeaturesResources.Find_References, supportsReferences: true);
+                    EditorFeaturesResources.Find_References, supportsReferences: true, CancellationToken.None);
 
-                var cancellationToken = context.CancellationToken;
-
-                // Kick off the work to do the actual finding on a BG thread.  That way we don'
-                // t block the calling (UI) thread too long if we happen to do our work on this
-                // thread.
-                await Task.Run(async () =>
+                try
+                {
+                    var cancellationToken = context.CancellationToken;
+
+                    // Kick off the work to do the actual finding on a BG thread.  That way we don'
+                    // t block the calling (UI) thread too long if we happen to do our work on this
+                    // thread.
+                    await Task.Run(async () =>
+                    {
+                        await FindReferencesAsync(symbolListItem, project, context).ConfigureAwait(false);
+                    }, cancellationToken).ConfigureAwait(false);
```

Comment:
```suggestion
                    await TaskScheduler.Default;
                    await FindReferencesAsync(symbolListItem, project, context).ConfigureAwait(false);
```

Reply:
out of curiosity, why?  isn't Task.Run by it's very nature siwtching us over to Default scheduler?

Reply:
Two reasons:

* It's much easier to read
* If it's already on the thread pool, this form is a NOP where the original will transition to yet another thread

Reply:
can you clarify thew first bullet.  it seems like extra stuff to read and understand.  

> If it's already on the thread pool

But... it will be on the threadpool.  That's the point of the Task.Run in the first place.  We use it precisely to get things on the threadpool.  So adding an extra line that will always NOP seems less clear or easy to read to me.